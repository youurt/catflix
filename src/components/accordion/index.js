import React, { useState, useContext, createContext } from 'react';
import {
  Container,
  Inner,
  Title,
  Item,
  Header,
  Body,
  Frame,
} from './styles/accordion';

const ToggleContext = createContext();

const Accordion = ({ children, ...restProps }) => {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
};
Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>;
};

Accordion.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Accordion.Item = ({ children, ...restProps }) => {
  const [toggleShow, setToggleShow] = useState(false);
  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

Accordion.Header = ({ children, ...restProps }) => {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);

  return (
    <Header
      onClick={() => setToggleShow((prevToggleShow) => !prevToggleShow)}
      {...restProps}
    >
      {children}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = ({ children, ...restProps }) => {
  const { toggleShow } = useContext(ToggleContext);

  return (
    <Body className={toggleShow ? 'open' : 'closed'} {...restProps}>
      <span>{children}</span>
    </Body>
  );
};

export default Accordion;
