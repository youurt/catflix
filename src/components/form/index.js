import React from 'react';
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from './styles/form';

const Form = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Form.Base = ({ children, ...restProps }) => {
  return <Base {...restProps}>{children}</Base>;
};

Form.Error = ({ children, ...restProps }) => {
  return <Error {...restProps}>{children}</Error>;
};

Form.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Form.Text = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

Form.TextSmall = ({ children, ...restProps }) => {
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

Form.Link = ({ children, ...restProps }) => {
  return <Link {...restProps}>{children}</Link>;
};

Form.Input = ({ children, ...restProps }) => {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = ({ children, ...restProps }) => {
  return <Submit {...restProps}>{children}</Submit>;
};

export default Form;
