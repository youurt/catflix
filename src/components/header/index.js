import React from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import { Background, Container, Logo, ButtonLink } from './styles/header';

const Header = ({ bg = true, children, ...restProps }) => {
  return bg ? <Background {...restProps}>{children}</Background> : children;
};

Header.Frame = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = ({ children, ...restProps }) => {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = ({ to, ...restProps }) => {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

export default Header;
