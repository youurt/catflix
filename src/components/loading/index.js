import React from 'react';
import { Spinner, LockBody, Picture, ReleaseBody } from './styles/loading';

const Loading = ({ src, ...restProps }) => {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid="loading-picture" />
    </Spinner>
  );
};

Loading.ReleaseBody = () => {
  return <ReleaseBody />;
};

export default Loading;
