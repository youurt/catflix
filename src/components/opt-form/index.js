import React from 'react';
import { Container, Input, Button, Text, Break } from './styles/opt-form';

const OptForm = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

OptForm.Input = ({ ...restProps }) => {
  return <Input {...restProps} />;
};

OptForm.Button = ({ children, ...restProps }) => {
  return (
    <Button {...restProps}>
      {children}
      <img src="/images/icons/chevron-right.png" alt="Try Now!" />
    </Button>
  );
};

OptForm.Text = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

OptForm.Break = function OptFormBreak({ ...restProps }) {
  return <Break {...restProps} />;
};

export default OptForm;
