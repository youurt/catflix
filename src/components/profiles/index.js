import React from 'react';
import { Container, Title, List, Picture, Name, Item } from './styles/profiles';

const Profiles = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Profiles.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Profiles.List = ({ children, ...restProps }) => {
  return <List {...restProps}>{children}</List>;
};

Profiles.User = ({ children, ...restProps }) => {
  return <Item {...restProps}>{children}</Item>;
};

Profiles.Name = ({ children, ...restProps }) => {
  return <Name {...restProps}>{children}</Name>;
};

Profiles.Picture = ({ src, ...restProps }) => {
  return (
    <Picture
      {...restProps}
      src={src ? `/images/users/${src}.png` : `/images/misc/loading.gif`}
    />
  );
};

export default Profiles;
