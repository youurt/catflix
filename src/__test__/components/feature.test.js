import React from 'react';
import { render } from '@testing-library/react';
import { Feature } from '../../components';

describe('<Feature/>', () => {
  it('should render the <Feature/> with with populated data', () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
        <Feature.SubTitle>Watch anywhere. Cancel any time.</Feature.SubTitle>
      </Feature>
    );

    expect(getByText('Unlimited films, TV programmes and more.')).toBeTruthy();
    expect(getByText('Watch anywhere. Cancel any time.')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the <Feature/> with just a title', () => {
    const { container, queryByText, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
      </Feature>
    );

    expect(getByText('Unlimited films, TV programmes and more.')).toBeTruthy();
    expect(queryByText('Watch anywhere. Cancel at any time.')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
