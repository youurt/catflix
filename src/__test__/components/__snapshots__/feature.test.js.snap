// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature/> should render the <Feature/> with just a title 1`] = `
<div
  class="feature__Container-sc-7i3nm1-0 gSHRFL"
>
  <h1
    class="feature__Title-sc-7i3nm1-1 cwuTdo"
  >
    Unlimited films, TV programmes and more.
  </h1>
</div>
`;

exports[`<Feature/> should render the <Feature/> with with populated data 1`] = `
<div
  class="feature__Container-sc-7i3nm1-0 gSHRFL"
>
  <h1
    class="feature__Title-sc-7i3nm1-1 cwuTdo"
  >
    Unlimited films, TV programmes and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-7i3nm1-2 kAeiwM"
  >
    Watch anywhere. Cancel any time.
  </h2>
</div>
`;
